File: app/src/main/java/hu/andras/daggersample/interactor/Feature2Interactor.java
package hu.andras.daggersample.interactor;

import hu.andras.daggersample.network.NetworkApi1;

/**
 * Created by Andras Nemeth on 2017. 06. 14..
 */

public class Feature2Interactor {

    private NetworkApi1 networkApi1;

    public Feature2Interactor(NetworkApi1 networkApi1) {
        this.networkApi1 = networkApi1;
        System.out.println("ddd Feature2Interactor created " + this);
    }

    public String getFeature2Stuff() {
        return "##" + networkApi1.getRequest() + "feature2 interactor: " + this + "##";
    }
}

File: app/src/main/java/hu/andras/daggersample/interactor/Feature2Interactor.kt
package hu.andras.daggersample.interactor

import hu.andras.daggersample.network.NetworkApi1

/**
 * Created by Andras Nemeth on 2017. 06. 14..
 */
class Feature2Interactor(private val networkApi1: NetworkApi1) {
    init {
        println("ddd Feature2Interactor created $this")
    }

    val feature2Stuff: String
        get() = "##" + networkApi1.request + "feature2 interactor: " + this + "##"
}
